client/app/bundles/course/assessment/components/AssessmentForm/index.tsx

```diff
@@ -178,7 +184,12 @@ const AssessmentForm = (props: AssessmentFormProps): JSX.Element => {
       encType="multipart/form-data"
       id="assessment-form"
       noValidate
-      onSubmit={handleSubmit((data) => onSubmit(data, setError))}
+      onSubmit={handleSubmit((data) =>
+        onSubmit(
+          { ...data, time_limit: data.has_time_limit ? data.time_limit : null },
```

Comment:
Why not add this when onSubmit is declared?

Reply:
moved the definition to the component several steps above it